rule/rule.go

```diff
@@ -1018,7 +1018,7 @@ func (r *Rule) IsEmpty(info KindInfo) bool {
 		return false
 	}
 	for k := range info.NonEmptyAttrs {
-		if _, ok := r.attrs[k]; ok {
+		if _, ok := r.attrs[k]; ok && !ShouldKeep(r.attrs[k].expr) {
```

Comment:
Doesn't this just make it harder for the function to return `false`, whereas the aim is to make it return `false` more often?